curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/serialization-of-a-user-object.english.md

```diff
@@ -18,12 +18,14 @@ passport.serializeUser((user, done) => {
   done(null, user._id);
 });
 passport.deserializeUser((id, done) => {
-  db.collection('users').findOne(
-    {_id: new ObjectID(id)},
+  myDB(async (client) => {
+    await client.db('database').collection('users').findOne(
+      { _id: new ObjectID(id) },
       (err, doc) => {
-        done(null, doc);
+        done(null, null);
       }
-  );
+    );
+  });
```

Comment:
Thank you, for pointing this out. This was an overlook of mine, as I did update the connection method, but forgot to change the lesson content.

Reply:
Sorry, I could have been more clear.

In my testing, I ran into problems with the following

```js
passport.deserializeUser((id, done) => {
  myDB(async (client) => {
    await client.db('database').collection('users').findOne(
      { _id: new ObjectID(id) },
      (err, doc) => {
        done(null, doc);
      }
    );
  });
});
```

after the Clean Up Your Project with Modules challenge. I would get the error `myDataBase is not a function` (renamed from `myDB` to `myDataBase` if going by the params in the instructions for the Clean Up Your Project with Modules challenge).

In the end I had to rewrite it as follows:

```js
passport.deserializeUser((id, done) => {
  myDataBase.findOne(
    {_id: new ObjectID(id)},
      (err, doc) => {
        done(null, doc);
      }
  );
});
```

Not sure if it's a mistake somewhere I made in the project, but my suggestion is just to have it as the following in this step to prevent possible errors in later challenges:

```js
passport.deserializeUser((id, done) => {
  myDB.findOne(
    {_id: new ObjectID(id)},
      (err, doc) => {
        done(null, doc);
      }
  );
});
```

Reply:
>after the Clean Up Your Project with Modules challenge.

Sorry, Kris. I still do not understand. [This lesson (the next one)](https://github.com/freeCodeCamp/freeCodeCamp/pull/39080/files#diff-e2fa44d297bf8e4f5ef85d88109b3fd5) expects:
>Be sure to remove the <code>myDB</code> code in deserializeUser and edit your <code>done(null, null)</code> to include the <code>doc</code>.

This is all done long before the _Clean Up Your Project_ lesson. If you take a look at the Repl.it, it should be similar. So, If you think this section is unclear as to what to do, then we definitely need to reword the instructions.

Reply:
Hi @Sky020 @Sky020, sorry, I must be missing something in these two steps.

Just to clarify, in Serialization of a User Object, learners will add the following to `server.js`:

```js
passport.serializeUser((user, done) => {
  done(null, user._id);
});
passport.deserializeUser((id, done) => {
  myDB(async client => {
    await client
      .db('database')
      .collection('users')
      .findOne({ _id: new ObjectID(id) }, (err, doc) => {
        done(null, null);
      });
  });
});
```

And they will comment out everything in `deserializeUser` except for `done(null, null);` so it doesn't throw an error.

Then in the next challenge, Implement the Serialization of a Passport User, learners set up their `MONGO_URI` string in `.env`, and encompass the code in what's provided in the description.

Also I see where they're supposed to refactor `deserializeUser`, which I must have missed the first time around. The only thing I did while running through the project before was to change `done(null, null);` to `done(null, doc);`.

So the current instructions, "Be sure to remove the `myDB` code in `deserializeUser` and edit your `done(null, null)` to include the doc", means to go from this in the previous challenge,

```js
passport.deserializeUser((id, done) => {
//   myDB(async client => {
//     await client
//       .db('database')
//       .collection('users')
//       .findOne({ _id: new ObjectID(id) }, (err, doc) => {
        done(null, null);
//       });
//   });
});
```

To this in Implement the Serialization of a Passport User:

```js
passport.deserializeUser((id, done) => {
  myDataBase.findOne({ _id: new ObjectID(id) }, (err, doc) => {
    if (err) return console.error(err);
      done(null, doc);
    });
});
```

Is that correct?

Reply:
@scissorsneedfoodtoo , yes that looks correct.

Reply:
@Sky020, great, thanks for confirming that.

I think it might be a bit too much of a jump between the two steps. "Be sure to remove the `myDB` code in `deserializeUser` and edit your `done(null, null)` to include the `doc`" almost sounds like learners are supposed to remove everything in `deserializeUser` except for `done(null, null)`, which they change to `done(null, doc)`.

One way we can ease the transition a bit is to change the example code in Serialization of a User Object to something like:

```js
passport.serializeUser((user, done) => {
  done(null, user._id);
});

passport.deserializeUser((id, done) => {
  myDB.findOne({ _id: new ObjectID(id) }, (err, doc) => {
    done(null, null);
  });
});
```

with the expectation they'll comment out everything but `done(null, null);`. None of the instructions should have to change in this step.

Then in the next step Implement the Serialization of a Passport User, we could modify the second to last sentence to read "Be sure to change `myDB` to `myDataBase` in `deserializeUser` and edit your `done(null, null)` to include the `doc`."

Does that sound like a good change? Or is there another way you'd like to break these steps up?

Reply:
I have done something similar; I decided not to have `myDB`, if it will be commented out and renamed, anyway.

Reply:
Awesome, thanks for making the changes. LGTM :+1: